### PR TITLE
bug fix in clone constructor of GsfElectron. 

### PR DESCRIPTION
**Pull request by shervin86**
New method to get regression correction without modifying the object, this method is needed in the ECALELF framework

### CODE REVIEW

File: RecoEgamma/EgammaTools/interface/SCEnergyCorrectorSemiParm.h [hunk 1]
```diff
@@ -33,7 +33,8 @@ class SCEnergyCorrectorSemiParm {
   ~SCEnergyCorrectorSemiParm(); 
   
   void setTokens(const edm::ParameterSet &iConfig, edm::ConsumesCollector &cc);
-  
+
+  std::pair<double,double> GetCorrections(const reco::SuperCluster &sc) const;
```
Comment (slava77) [inline RecoEgamma/EgammaTools/interface/SCEnergyCorrectorSemiParm.h]: start local variables and data members with lower case letter. Capitalization is reserved for class names.

Comment (shervin86) [inline RecoEgamma/EgammaTools/interface/SCEnergyCorrectorSemiParm.h]: implemented :-)

**Comment by shervin86** [comment]
It would be good to have the PR ported to 80X to get it in the next patch release. If you agree I can make the PR

File: RecoEgamma/EgammaTools/src/SCEnergyCorrectorSemiParm.cc [hunk 2]
```diff
@@ -294,9 +297,22 @@ void SCEnergyCorrectorSemiParm::modifyObject(reco::SuperCluster &sc) {
...
-    sc.setCorrectedEnergy(ecor);
-  }  
+//--------------------------------------------------------------------------------------------------
+void SCEnergyCorrectorSemiParm::modifyObject(reco::SuperCluster &sc) {
+  
+	std::pair<double, double> cor = getCorrections(sc);
+	if(cor.first<0 || cor.second < 0) return;
+	sc.setEnergy(cor.first);
+	sc.setCorrectedEnergy(cor.first);
+	sc.setCorrectedEnergyUncertainty(cor.second);
```
Comment (slava77) [inline RecoEgamma/EgammaTools/src/SCEnergyCorrectorSemiParm.cc]: if I'm not mistaken, this changes the logic.
Previously, setCorrectedEnergyUncertainty was not called for the HLT case, while now it is set to 0.

Comment (shervin86) [inline RecoEgamma/EgammaTools/src/SCEnergyCorrectorSemiParm.cc]: It's true. In my understanding the energy uncertainty is not used at HLT. If it is not set by this class, then it should have a default value.
I can add an if(! isHLT) at L316 to preserve the value.

Comment (slava77) [inline RecoEgamma/EgammaTools/src/SCEnergyCorrectorSemiParm.cc]: default ("uninitialized") energy uncertainty is -1.
I think we better keep it like that  at -1 and not change it to 0 for cases where we think the value is not used.

**Comment by slava77** [comment]
@cmsbuild please test

@shervin86 if you need this in other releases, it's now more practical to make PRs for these other releases.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/16752 54298e5
- code changes look OK: mostly technical update together with a bugfix affecting analysis setup (not standard workflows)
- jenkins tests pass and comparisons with baseline show no differences

**Comment by davidlange6** [comment]
+1

**Pull request merged**